ERCS/erc-7752.md

```diff
@@ -0,0 +1,607 @@
+   */
+  function name() external view returns (string memory);
+
+  /**
+   * @dev Returns the address of the onchainID of the token.
+   * the onchainID of the token gives all the information available
+   * about the token and is managed by the token issuer or his agent.
+   */
+  function onchainID() external view returns (address);
+
+  /**
+   * @dev Returns the symbol of the token, usually a shorter version of the
+   * name.
+   */
+  function symbol() external view returns (string memory);
+
+  /**
+   * @dev Returns the TREX version of the token.
+   * current version is 3.0.0
+   */
+  function version() external view returns (string memory);
+
+  /**
+   *  @dev Returns the Identity Registry linked to the token
+   */
+  function identityRegistry() external view returns (IIdentityRegistry);
+
+  /**
+   *  @dev Returns the Compliance contract linked to the token
+   */
+  function compliance() external view returns (IModularCompliance);
+
+  /**
+   * @dev Returns true if the contract is paused, and false otherwise.
+   */
+  function paused() external view returns (bool);
+
+  /**
+   *  @dev Returns the freezing status of a wallet
+   *  if isFrozen returns `true` the wallet is frozen
+   *  if isFrozen returns `false` the wallet is not frozen
+   *  isFrozen returning `true` doesn't mean that the balance is free, tokens could be blocked by
+   *  a partial freezeSecurity or the whole token could be blocked by pause
+   *  @param _userAddress the address of the wallet on which isFrozen is called
+   */
+  function isFrozen(address _userAddress) external view returns (bool);
+}
+```
+
+## Rationale
+
+Assigning a unique `securityId` to all issuances of equity tokens is the industry-accepted practice for assignment of unique vesting conditions, audit, and cost-basis tracking purposes. To keep the implementation under the max size limit, vesting conditions and governing documents are to be separate contracts with related data accessed by specifying a `securityId`.
+
+## Backwards Compatibility
+
+Equity tokens are not backwards compatible with [ERC-20](./eip-20.md) nor [ERC-3643](./eip-3643.md).
+
+This standard introduces breaking changes to [ERC-3643](./eip-3643.md) by emphasizing each security issuance rather than fungible token balances. Methods like `freezePartialTokens` have been removed in favor of `freezeSecurity`. An additional `securityId` param is required for any method that modifies an account’s equity balance.
+
+In addition to `mint` and `burn` [ERC-20](./eip-20.md) breaking changes, this standard introduces:
+
+- the `shareClass` and `uri` parameters to the `issue` method
+- a `securityId` parameter to `cancel`, `transfer`, and `transferFrom`, methods
+- a `securityIds` parameter to corresponding batch methods
+- a `shareClass` param to the `approve`, `increaseAllowance`, and `decreaseAllowance` methods
+- corresponding event signatures.
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
```suggestion
Needs discussion. <!-- TODO -->
```

If you use an HTML-style comment, the linter will ensure you replace it before going into Review.